src/Agda2Hs/Compile/Type.hs

```diff
@@ -193,5 +197,7 @@ compileKeptTeleBind x t = do
 compileKind :: Type -> Maybe (Hs.Kind ())
 compileKind t = case unEl t of
   Sort (Type _) -> pure (Hs.TyStar ())
-  Pi a b -> Hs.TyFun () <$> compileKind (unDom a) <*> compileKind (unAbs b)
-  _ -> Nothing
+  Pi a b
+    | usableModality a  -> Hs.TyFun () <$> compileKind (unDom a) <*> compileKind (unAbs b)
```

Comment:
Same here: use `keepArg` and `underAbstraction`

Reply:
Oh, sorry; I've missed that... I'll do that soon; this is a `Pi`, so I think I won't have problems with it.

Reply:
I think it's done now:) I hope I understood it well.